### DISCUSSION TITLE
Why do I get Identifier Expected , and Statement expected here

### DISCUSSION THREAD

**Discussion by Meriyemelhajoui** [microsoft/typespec#3437] [Q&A]
Why do I get Identifier Expected , and Statement expected here :
```
model Person{
name?: string;
adress?: Address_1_0_0; // this is a seperate model defined 
age?: integer;

   
   ...AdditionalProperties<never>;
}
``` @allenjzhang @Timothee Guerin

**Comment by bterlson** [comment]
Nothing you pasted here has a syntax error. Can you include more context? A [playground ](https://typespec.io/playground) repro would be helpful too!

**Reply by Meriyemelhajoui** [reply]
token-expected: Identifier expected.
> 47 | adress?: Address_1_0_0;
error token-expected: Statement expected.
> 48 | name?: string;

**Reply by bterlson** [reply]
If I had to guess you have a syntax error before this declaration somewhere, likely another declaration that is missing its closing brace?

**Reply by Meriyemelhajoui** [reply]
I've checked but I don't have any syntax error , I tried to remove this piece of code , and it works so basically the error related only to this section

**Reply by bterlson** [reply]
If you remove everything but this piece of code, do you still have a syntax error? I'm trying to figure out why [this](https://typespec.io/playground?c=bW9kZWwgUGVyc29uew0KICBuYW1lPzogc3RyaW5nO8QSYWRyZXNzPzogQWTFCl8xXzBfMDsgLy8gdGhpcyBpcyBhIHNlcGVyYXRlIMZRZGVmaW5lZCDFQGfEUWludGVnZXLFUi4uLkFkZGl0aW9uYWxQcm9wZXJ0aWVzPG5ldmVyPjsNCn0NCg0KxkrVKVQ%2BIHsgfcsj6gCYIHt9&e=%40typespec%2Fopenapi3&options=%7B%7D) works but what you're seeing doesn't...

**Reply by Meriyemelhajoui** [reply]
Yes , I tried to remove all code , except this one , and I still get the same errors :)

**Reply by bterlson** [reply]
What happens if you take the code that you have left after eliminating all the other code and paste it into the playground I linked? If you get the error, can you link me? (Hit the save button, and the url should update to something you can share).

**Reply by Meriyemelhajoui** [reply]
Sure , I got the same errors , when I paste it into the playground . 
[Link to playground](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7CsUBbW9kZWwgRXhwcmVzc2lvbkZfMV8wXzB7CmXJEz86IHN0cmluZzsKb3A%2FOiBDb21wYXJpc29uX09wxi87CuQAoMspZmlsdGVySUTLEyAgCn0KCg%3D%3D&e=%40typespec%2Fjson-schema&options=%7B%7D)

**Reply by bterlson** [reply]
Ahh, the issue is that you have a property named "op", which is a keyword. You have to surround it in backticks [like so](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7CsUBbW9kZWwgRXhwcmVzc2lvbkZfMV8wXzB7CmXJEz86IHN0cmluZzsKYG9wYD86IENvbXBhcmlzb25fT3DGMTsK5ACiyytmaWx0ZXJJRMsTICAKfQoK&e=%40typespec%2Fjson-schema&options=%7B%7D).

**Reply by Meriyemelhajoui** [reply]
Yees , it works now , thank you I've spent 3 hours blocked on that , is there any page that highlight the reserved keywords ? and also Syntax for naming properties , as my knowledge we can't start a property with a number and also with bracket ...

**Reply by bterlson** [reply]
Sorry to hear! We don't have great docs on that, it's a good point that we should add it! If you can read formal grammar notation, you can find the spec [here](https://github.com/microsoft/typespec/blob/main/packages/spec/src/spec.emu.html#L52). In general we use the Unicode [UAX31-R2](http://www.unicode.org/reports/tr31/#R2) for identifiers, so in general word characters in any language, emoji, $ and _ are fine. Other symbols are generally not fine. Numbers can't start an identifier but can be present within an identifier.

**Reply by Meriyemelhajoui** [reply]
Alright , thank you so much
